src/raz_meta.rs

```diff
@@ -0,0 +1,65 @@
+//! Available meta data for the inc_gauged_raz
+//! 
+//! This meta data will be used when focusing into the Raz
+
+use std::fmt::Debug;
+use std::hash::Hash;
+use adapton::engine::Name;
+
+pub trait RazMeta<E>: Sized+Debug+Clone+Eq+Hash {
+	type Index;
+
+	/// create meta data for an empty branch
+	fn from_none() -> Self;
+	/// create meta data from a leaf vec
+	fn from_vec(vec: &Vec<E>) -> Self;
+	/// create meta data from the pair of meta data in branches
+	///
+	/// The names passed into here are USED in the tree that it
+	/// rebuilds the meta data for, and should not be used again to
+	/// name arts
+	fn from_meta(l: &Self, r: &Self, l: u32, n: Option<Name>) -> Self;
+	/// choose a branch and create an adjusted index for that branch
+	fn choose_side(l: &Self, r: &Self, index: &Self::Index) -> SideChoice<Self::Index>;
```

Comment:
Perhaps rename this `navigate`?  The doc comment didn't make it clear to me initially that this function navigates the tree structure in the RAZ, to find a particular `Self::Index`.  However, after I read the impl for `Count`, the role of `choose_side` became clear.